plasmapy/simulation/particle_tracker.py

```diff
@@ -0,0 +1,439 @@
+            An explicitly set timestep in units convertible to seconds.
+            Setting this optional keyword overrules the adaptive time step
+            capability and forces the use of this timestep throughout. If a tuple
+            of timesteps is provided, the adaptive timestep will be clamped
+            between the first and second values.
+
+        field_weighting : str
+            String that selects the field weighting algorithm used to determine
+            what fields are felt by the particles. Options are:
+
+            * 'nearest neighbor': Particles are assigned the fields on
+                the grid vertex closest to them.
+            * 'volume averaged' : The fields experienced by a particle are a
+                volume-average of the eight grid points surrounding them.
+
+            The default is 'volume averaged'.
+
+        Returns
+        -------
+        None
+
+        """
+
+        # Load and validate inputs
+        field_weightings = ["volume averaged", "nearest neighbor"]
+        if field_weighting in field_weightings:
+            self.field_weighting = field_weighting
+        else:
+            raise ValueError(
+                f"{field_weighting} is not a valid option for ",
+                "field_weighting. Valid choices are",
+                f"{field_weightings}",
+            )
+
+        # Check to make sure particles have already been generated
+        if not hasattr(self, "x"):
+            raise ValueError(
+                "Either the create_particles or load_particles method must be "
+                "called before running the particle tracing algorithm."
+            )
+
+        self.nparticles_tracked = self.x.shape[0]
+
+        # By default, set dt as an infinite range (auto dt with no restrictions)
+        self.dt = np.array([0.0, np.inf]) * u.s if dt is None else dt
+        self.dt = (self.dt).to(u.s).value
+
+        # Store a copy of the initial velocity distribution in memory
+        # This will be used later to calculate the maximum deflection
+        self.v_init = np.copy(self.v)
+
+        # Create flags for tracking when particles during the simulation
+        # on_grid -> zero if the particle is off grid, 1
+        # shape [nparticles, ngrids]
+        self.on_grid = np.zeros([self.nparticles_tracked, self.num_grids])
+
+        # Initialize a "progress bar" (really more of a meter)
+        # Setting sys.stdout lets this play nicely with regular print()
+        pbar = tqdm(
+            initial=0,
+            total=self.nparticles_tracked + 1,
+            disable=not self.verbose,
+            desc="Particles on grid",
+            unit="particles",
+            bar_format="{l_bar}{bar}{n:.1e}/{total:.1e} {unit}",  # noqa: FS003
+            file=sys.stdout,
+        )
+
+        # Push the particles until the stop condition is satisfied
+        # (no more particles on the simulation grid)
+        while True:  # not stop_condition():
```

Comment:
Since we aren't defining the radiography tracer to be a subclass of the general tracer, I ran into some issues with passing arguments to the `stop_condition` method. How would you recommend I proceed?